### PR TITLE
Navigation drawer

### PR DESCRIPTION
**Pull request by solandmedotru**
Can you see to style.xml ?
If i write style name="Theme.AppCompat.Light.NoActionBar" parent="@style/Theme.AppCompat.Light" its worked, but if i write style name="AppTheme" parent="Theme.AppCompat.Light.NoActionBar" then get some error You need to use a Theme.AppCompat theme

### CODE REVIEW

**Comment by rovkinmax** [comment]
just set theme for coordinator layout
`coordinatorLayout(theme = R.style.AppTheme_NoActionBar)`

**Comment by atetc** [comment]
@solandmedotru please pull changes and solve conflicts

**Comment by solandmedotru** [comment]
Я не вижу никаких конфликтов. Все исправления нужны и на что там ругается CI я не понимаю. Может глянете кто разбирается в этом?

**Comment by solandmedotru** [comment]
I don't known what is wrong in CI. Who give me some hint, direction to look

**Comment by atetc** [comment]
@solandmedotru problem with Unit tests, will investigate

Please write Unit tests for your classes (you can use Robolectric & Mockito)

**Comment by solandmedotru** [comment]
@atetc Robolectric does not support API level 24. Maybe change targetSdk on 19-23?
If i change support lib v '24.2.1' to '24.2.0' and TargetSDK to 22 Robolectric working fine. It have issue on google https://code.google.com/p/android/issues/detail?id=222597 and https://code.google.com/p/android/issues/detail?id=222597

**Comment by solandmedotru** [comment]
@atetc I added some simple tests (not null toolbar and drawer). But CI is red. Maybe add more test.

File: dependencies.gradle [hunk 1]
```diff
@@ -3,24 +3,35 @@ ext.versions = [
...
+        appSupport   : "com.android.support:support-v4:$versions.support",
 
-        junit               : "junit:junit:$versions.junit",
-        kotlinJunit         : "org.jetbrains.kotlin:kotlin-test-junit:$kotlin_version",
-        mockitoCore         : "org.mockito:mockito-core:$versions.mockito"
+        junit        : "junit:junit:$versions.junit",
+        kotlinJunit  : "org.jetbrains.kotlin:kotlin-test-junit:$versions.kotlin",
+        mockitoCore  : "org.mockito:mockito-core:$versions.mockito",
+        espresso     : "com.android.support.test.espresso:espresso-core:$versions.espresso",
+        robolectric  : "org.robolectric:robolectric:3.1.2"
```
Comment (atetc) [inline dependencies.gradle]: Extract version to versions list

File: dependencies.gradle [hunk 2]
```diff
@@ -3,24 +3,35 @@ ext.versions = [
         name                : '0.1',
 
         minSdk              : 19,
-        targetSdk           : 24,
+        targetSdk           : 22,
```
Comment (atetc) [inline dependencies.gradle]: What is the reason of this downgrade?

Comment (solandmedotru) [reply dependencies.gradle]: @atetc Robolectric does not support API level 24.
If i change support lib v '24.2.1' to '24.2.0' and TargetSDK to 22 Robolectric working fine. It have issue on google https://code.google.com/p/android/issues/detail?id=222597 and https://code.google.com/p/android/issues/detail?id=222597

Comment (solandmedotru) [reply dependencies.gradle]: Or in Config hard use (constants = BuildConfig::class, manifest = "AndroidManifest.xml", sdk = intArrayOf(Build.VERSION_CODES.LOLLIPOP))

File: app/src/main/res/values/ids.xml [hunk 3]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+    <item name="drawer" type="id" />
+    <item name="toolbar" type="id" />
```
Comment (atetc) [inline app/src/main/res/values/ids.xml]: Why we need this ids if we don't use finViewById()?

File: app/src/test/kotlin/com/androidresid/androidapp/KotlinJunitTest.kt [hunk 4]
```diff
@@ -1,15 +1,15 @@
...
 import kotlin.test.assertTrue
 
 class KotlinJunitTest {
 
     @Test
     fun firstTest() {
-        val mainActivity = MainActivity();
-        assertTrue(mainActivity.isCoverageTestWorks());
+        val mainActivity = MainActivity()
+        assertTrue(mainActivity.isCoverageTestWorks())
```
Comment (atetc) [inline app/src/test/kotlin/com/androidresid/androidapp/KotlinJunitTest.kt]: We should remove it

File: app/src/test/kotlin/com/androidresid/androidapp/KotlinJunitTest.kt [hunk 5]
```diff
@@ -1,15 +1,15 @@
...
+        val mainUI = MainUI()
+        assertTrue(mainUI.isCoverageTestWorks())
```
Comment (atetc) [inline app/src/test/kotlin/com/androidresid/androidapp/KotlinJunitTest.kt]: We should remove it

File: app/src/test/kotlin/com/androidresid/androidapp/MainActivityTest.kt [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+    @Test
+    fun isShouldDisplayTitle(){
+        assertTrue(activity.title.toString() == ("AndroidResId"))
+    }
+
+    @Test
+    fun itShouldDisplayToolbar(){
+        val toolbar = activity.findViewById(R.id.toolbar)
+        assertNotNull(toolbar)
```
Comment (atetc) [inline app/src/test/kotlin/com/androidresid/androidapp/MainActivityTest.kt:41]: Can we also check is toolbar visible?

File: app/src/test/kotlin/com/androidresid/androidapp/MainActivityTest.kt [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+    }
+    @Test
+    fun itShouldDisplayNavigationDrawer(){
+        val drawer = activity.findViewById(R.id.drawer)
+        assertNotNull(drawer)
```
Comment (atetc) [inline app/src/test/kotlin/com/androidresid/androidapp/MainActivityTest.kt:48]: Can we also check is drawer visible?

**Comment by atetc** [comment]
solandmedotru jacoco coverage report temporary switched off

File: app/src/test/kotlin/com/androidresid/androidapp/MainActivityTest.kt [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+    }
+    @Test
+    fun itShouldMenuItemSetting(){
+
+    }
```
Comment (atetc) [inline app/src/test/kotlin/com/androidresid/androidapp/MainActivityTest.kt:58]: Why does it empty?

File: app/src/test/kotlin/com/androidresid/androidapp/MainActivityTest.kt [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+    @Test
+    fun itShouldDisplayNavigationDrawer(){
+        val drawer = activity.findViewById(R.id.drawer)
+        assertNotNull(drawer)
+        assertTrue(drawer.visibility == View.VISIBLE)
+    }
+    @Test
+    fun itShouldMenuItemSetting(){
+        RoboMenuItem(R.id.action_settings).isVisible
```
Comment (atetc) [inline app/src/test/kotlin/com/androidresid/androidapp/MainActivityTest.kt]: Assertions missed

File: app/src/test/kotlin/com/androidresid/androidapp/MainActivityTest.kt [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+
+@RunWith(RobolectricTestRunner::class)
+@Config(constants = BuildConfig::class, manifest = "AndroidManifest.xml", sdk = intArrayOf(Build.VERSION_CODES.LOLLIPOP))
+class MainActivityTest {
+
+    val activity = Robolectric.buildActivity(MainActivity::class.java)
+            .create()
+            .start()
+            .resume()
+            .visible()
```
Comment (atetc) [inline app/src/test/kotlin/com/androidresid/androidapp/MainActivityTest.kt]: Can we use setup(); here?

**Pull request merged**